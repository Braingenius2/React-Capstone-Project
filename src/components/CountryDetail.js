import PropTypes from 'prop-types';

const CountryDetail = ({ title, amount }) => (
  <li>
    <div>
      <h3>{ title }</h3>
    </div>
    <div>
      <p>{ amount }</p>
    </div>
  </li>
);

CountryDetail.propTypes = {
  title: PropTypes.string.isRequired,
  amount: PropTypes.string.isRequired,
};

export default CountryDetail;
