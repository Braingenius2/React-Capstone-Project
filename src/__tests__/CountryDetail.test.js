import '@testing-library/jest-dom/extend-expect';
import { render, screen } from '@testing-library/react';
import CountryDetail from '../components/CountryDetail';

describe('The CountryDetail component', () => {
  test('should render correctly into the DOM', () => {
    const { CountryDetailComponent } = render(
      <CountryDetail title="cases" amount="9,106" />,
    );
    expect(CountryDetailComponent).toMatchSnapshot();
  });

  test('should render the text: recovered and the number of recovered people: 36,366', () => {
    render(
      <CountryDetail title="recovered" amount="36,366" />,
    );
    const CountryDetailTitle = screen.getByText(/recovered/i);
    const CountryDetailCases = screen.getByText(/36,366/i);

    expect(CountryDetailTitle).toBeInTheDocument();
    expect(CountryDetailCases).toBeInTheDocument();
  });
});
