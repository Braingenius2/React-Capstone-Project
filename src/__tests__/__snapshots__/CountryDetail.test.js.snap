// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The CountryDetail component should render correctly into the DOM 1`] = `undefined`;
